pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -226,6 +228,89 @@ var _ = Describe("podNIC", func() {
+				fakeAddr1 := netlink.Addr{IPNet: address1}
+				fakeAddr2 := netlink.Addr{IPNet: address2}
+				mockNetwork.EXPECT().ReadIPAddressesFromLink(primaryPodInterfaceName).Return(fakeAddr1.IP.String(), fakeAddr2.IP.String(), nil)
+				mockNetwork.EXPECT().IsIpv4Primary().Return(true, nil).Times(1)
+				mockPodNetworkConfigurator.EXPECT().DiscoverPodNetworkInterface(podnic.podInterfaceName).Times(1)
+				mockPodNetworkConfigurator.EXPECT().GenerateDomainIfaceSpec().Return(api.Interface{})
+				mockPodNetworkConfigurator.EXPECT().GenerateNonRecoverableDHCPConfig().Return(&cache.DHCPConfig{}).Times(1)
+			})
+			Context("and prepare networking is fine at phase1 calling", func() {
+				BeforeEach(func() {
+					mockPodNetworkConfigurator.EXPECT().PreparePodNetworkInterface().Times(1)
+					err := podnic.PlugPhase1()
+					Expect(err).ToNot(HaveOccurred())
+				})
+				It("should transition to PodIfaceNetworkingPrepared state", func() {
+					obtainedState, err := podnic.state()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(obtainedState).To(Equal(cache.PodIfaceNetworkingPrepared))
+				})
+			})
+			Context("and prepare networking fails at phase1 calling", func() {
+				BeforeEach(func() {
+					mockPodNetworkConfigurator.EXPECT().PreparePodNetworkInterface().Times(1).Return(fmt.Errorf("podnic_test: forced PreparePodNetworkInterface failure"))
+					err := podnic.PlugPhase1()
+					Expect(err).To(HaveOccurred())
+				})
+				It("should set state to PodIfaceNetworkingUnknown", func() {
+					obtainedState, err := podnic.state()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(obtainedState).To(Equal(cache.PodIfaceNetworkingUnknown))
+				})
+			})
+		})
+		When("state is PodIfaceNetworkingUnknown", func() {
+			var podnic *podNIC
+			BeforeEach(func() {
+				var err error
+				vmi := newVMIMasqueradeInterface("testnamespace", "testVmName", masqueradeCidr, masqueradeIpv6Cidr)
+				podnic, err = newPhase1PodNICWithMocks(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				podnic.setState(cache.PodIfaceNetworkingUnknown)
+			})
+			Context("and phase1 is called", func() {
+				It("should fail with CriticalNetworkError", func() {
+					err := podnic.PlugPhase1()
+					Expect(err).To(HaveOccurred(), "SetupPhase1 should return an error")
+
+					_, ok := err.(*neterrors.CriticalNetworkError)
+					Expect(ok).To(BeTrue(), "SetupPhase1 should return an error of type CriticalNetworkError")
+				})
+			})
+		})
+		When("state is PodIfaceNetworkingPrepared", func() {
+			var podnic *podNIC
+			BeforeEach(func() {
+				var err error
+				vmi := newVMIMasqueradeInterface("testnamespace", "testVmName", masqueradeCidr, masqueradeIpv6Cidr)
+				podnic, err = newPhase1PodNICWithMocks(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				podnic.setState(cache.PodIfaceNetworkingPrepared)
+			})
+			Context("and phase1 is called", func() {
+				It("should do be a noop", func() {
+					Expect(podnic.PlugPhase1()).To(Succeed())
```

Comment:
how do you know if this a no op ? It could have ran the real method for all you know from the API.

Reply:
This delegates on mocks expectations since we don't set any they will fail if the code is run.

Reply:
I'm just complaining that the expectation string does not make sense to me.